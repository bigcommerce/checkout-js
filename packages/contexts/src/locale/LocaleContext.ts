import type { CurrencyService, LanguageService } from '@bigcommerce/checkout-sdk';
import { createContext } from 'react';

export interface LocaleContextType {
    language: LanguageService;
    date?: {
        inputFormat: string;
    };
    currency?: CurrencyService;
}

export const LocaleContext = createContext<LocaleContextType | undefined>(undefined);
