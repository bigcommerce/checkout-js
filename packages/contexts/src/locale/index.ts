export { LocaleContext, type LocaleContextType } from './LocaleContext';
export { default as LocaleProvider } from './LocaleProvider';
export { useLocale } from './useLocale';
