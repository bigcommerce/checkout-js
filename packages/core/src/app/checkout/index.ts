export { RenderCheckout, RenderCheckoutOptions } from './renderCheckout';
export { default as withCheckout, WithCheckoutProps } from './withCheckout';
export { default as CheckoutSupport } from './CheckoutSupport';
export { default as NoopCheckoutSupport } from './NoopCheckoutSupport';
export { default as navigateToOrderConfirmation } from './navigateToOrderConfirmation';
