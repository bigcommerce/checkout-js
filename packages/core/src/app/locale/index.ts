export { default as LocaleContext, LocaleContextType } from './LocaleContext';
export { default as createLocaleContext } from './createLocaleContext';
export { default as getDefaultTranslations } from './getDefaultTranslations';
export { default as getLanguageService, InitializeLanguageService } from './getLanguageService';
export { default as masterpassFormatLocale } from './masterpassFormatLocale';
export { default as withCurrency, WithCurrencyProps } from './withCurrency';
export { default as withLanguage, WithLanguageProps } from './withLanguage';
export { default as withDate, WithDateProps } from './withDate';
export { default as LanguageWindow, isLanguageWindow } from './LanguageWindow';
export { default as LocaleProvider } from './LocaleProvider';
export { default as TranslatedHtml, TranslatedHtmlProps } from './TranslatedHtml';
export { default as TranslatedLink, TranslatedLinkProps } from './TranslatedLink';
export { default as TranslatedString, TranslatedStringProps } from './TranslatedString';
