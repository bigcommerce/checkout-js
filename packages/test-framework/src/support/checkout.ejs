<!DOCTYPE html>
<html>
    <head>
        <title>Local Testing Environment</title>
        <base href="/" target="_top"/>
        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
        <link href="https://fonts.googleapis.com/css?family=Karla:400|Montserrat:400,500,700&display=swap" rel="stylesheet" />
    </head>
    <body>
    <header class="checkoutHeader optimizedCheckout-header">
        <div class="checkoutHeader-content">
            <h1 class="is-srOnly">Checkout</h1>
            <h2 class="checkoutHeader-heading">
                <a class="checkoutHeader-link" href="<%= storeUrl %>/">
                    <span class="header-logo-text">Local Testing Environment</span>
                </a>
            </h2>
        </div>
    </header>
    <div class="ng-checkout-container" id="micro-app-ng-checkout">
        <div class="ng-checkout-view" data-ui-view></div>
    </div>
    <script type="text/javascript">
        window.checkoutConfig = {
            checkoutId: '<%= checkoutId %>',
            containerId: 'micro-app-ng-checkout',
        };
    </script>
    <script src="<%= localhostUrl %>/auto-loader.js" type="text/javascript"></script>
    <script type="text/javascript">
        (function() {
            window.checkout.renderCheckout({
                checkoutId: '<%= checkoutId %>',
                containerId: 'micro-app-ng-checkout',
                sentryConfig: null,
            })
        })();
    </script>
    </body>
</html>
