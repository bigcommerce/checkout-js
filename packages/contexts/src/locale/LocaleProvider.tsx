import {
    type CheckoutService,
    createCurrencyService,
    type LanguageService,
    type StoreConfig,
} from '@bigcommerce/checkout-sdk/essential';
import { memoizeOne } from '@bigcommerce/memoize';
import React, { type ReactNode, useEffect, useMemo, useState } from 'react';

import { LocaleContext } from './LocaleContext';

export interface LocaleProviderProps {
    languageService: LanguageService;
    checkoutService: CheckoutService;
    children?: ReactNode;
}

const LocaleProvider: React.FC<LocaleProviderProps> = ({
    checkoutService,
    children,
    languageService,
}) => {
    const [config, setConfig] = useState<StoreConfig | undefined>();

    const getContextValue = memoizeOne((storeConfig?: StoreConfig) => {
        return {
            currency: storeConfig ? createCurrencyService(storeConfig) : undefined,
            date: storeConfig
                ? {
                      inputFormat: storeConfig.inputDateFormat,
                  }
                : undefined,
            language: languageService,
        };
    });

    useEffect(() => {
        const unsubscribe = checkoutService.subscribe(
            ({ data }) => {
                setConfig(data.getConfig());
            },
            ({ data }) => data.getConfig(),
        );

        return () => {
            unsubscribe();
        };
    }, [checkoutService]);

    const contextValue = useMemo(() => getContextValue(config), [config, getContextValue]);

    return <LocaleContext.Provider value={contextValue}>{children}</LocaleContext.Provider>;
};

export default LocaleProvider;
