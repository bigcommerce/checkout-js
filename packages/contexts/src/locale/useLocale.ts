import { useContext } from 'react';

import { LocaleContext } from './LocaleContext';

export function useLocale() {
    const context = useContext(LocaleContext);

    if (!context) {
        throw new Error('useLocale must be used within a LocaleContextProvider');
    }

    return context;
}
