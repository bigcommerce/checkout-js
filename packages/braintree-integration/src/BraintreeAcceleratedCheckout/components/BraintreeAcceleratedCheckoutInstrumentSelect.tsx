import { CardInstrument } from '@bigcommerce/checkout-sdk';
import { FieldProps } from 'formik';
import React, { FunctionComponent, useEffect } from 'react';

import { usePaymentFormContext } from '@bigcommerce/checkout/payment-integration-api';
import { DropdownTrigger } from '@bigcommerce/checkout/ui';

import BraintreeAcceleratedCheckoutInstrumentMenu from './BraintreeAcceleratedCheckoutInstrumentMenu';
import BraintreeAcceleratedCheckoutInstrumentSelectButton from './BraintreeAcceleratedCheckoutInstrumentSelectButton';
import PoweredByPaypalConnectLabel from './PoweredByPaypalConnectLabel';

interface BraintreeAcceleratedCheckoutInstrumentSelectProps extends FieldProps<string> {
    instruments: CardInstrument[];
    selectedInstrument?: CardInstrument;
    onSelectInstrument(id: string): void;
    onUseNewInstrument(): void;
}

const BraintreeAcceleratedCheckoutInstrumentSelect: FunctionComponent<
    BraintreeAcceleratedCheckoutInstrumentSelectProps
> = ({ field, selectedInstrument, instruments, onSelectInstrument, onUseNewInstrument }) => {
    const { paymentForm } = usePaymentFormContext();

    const updateFieldValue = (instrumentId?: string) => {
        paymentForm.setFieldValue('instrumentId', instrumentId);
    };

    useEffect(() => {
        updateFieldValue(selectedInstrument?.bigpayToken);

        return () => updateFieldValue();
    }, [selectedInstrument]);

    return (
        <div className="instrumentSelect" data-test="instrument-select">
            <DropdownTrigger
                dropdown={
                    <BraintreeAcceleratedCheckoutInstrumentMenu
                        instruments={instruments}
                        onSelectInstrument={onSelectInstrument}
                        onUseNewInstrument={onUseNewInstrument}
                        selectedInstrument={selectedInstrument}
                    />
                }
            >
                <BraintreeAcceleratedCheckoutInstrumentSelectButton
                    instrument={selectedInstrument}
                />

                <input type="hidden" {...field} />
            </DropdownTrigger>

            {selectedInstrument && <PoweredByPaypalConnectLabel />}
        </div>
    );
};

export default BraintreeAcceleratedCheckoutInstrumentSelect;
