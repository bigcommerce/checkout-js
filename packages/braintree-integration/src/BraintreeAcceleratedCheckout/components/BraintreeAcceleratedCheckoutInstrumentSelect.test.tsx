import { render, screen } from '@testing-library/react';
import { Field, FieldProps, Formik } from 'formik';
import { noop } from 'lodash';
import React from 'react';

import { createLocaleContext, LocaleContext } from '@bigcommerce/checkout/locale';
import { PaymentFormContext } from '@bigcommerce/checkout/payment-integration-api';
import {
    getCardInstrument,
    getPaymentFormServiceMock,
    getStoreConfig,
} from '@bigcommerce/checkout/test-utils';

import BraintreeAcceleratedCheckoutInstrumentSelect from './BraintreeAcceleratedCheckoutInstrumentSelect';

jest.mock('./PoweredByPaypalConnectLabel', () => () => (
    <div data-test="powered-by-paypal-connect-label">PoweredByPaypalConnectLabel</div>
));

describe('BraintreeAcceleratedCheckoutInstrumentSelect', () => {
    it('renders instrument select', () => {
        const cardInstrument = getCardInstrument();

        const view = render(
            <Formik initialValues={{}} onSubmit={noop}>
                <LocaleContext.Provider value={createLocaleContext(getStoreConfig())}>
                    <PaymentFormContext.Provider
                        value={{ paymentForm: getPaymentFormServiceMock() }}
                    >
                        <Field
                            name="instrumentId"
                            render={(field: FieldProps<string>) => (
                                <BraintreeAcceleratedCheckoutInstrumentSelect
                                    instruments={[cardInstrument]}
                                    onSelectInstrument={noop}
                                    onUseNewInstrument={noop}
                                    selectedInstrument={cardInstrument}
                                    {...field}
                                />
                            )}
                        />
                    </PaymentFormContext.Provider>
                </LocaleContext.Provider>
            </Formik>,
        );

        expect(view).toMatchSnapshot();
    });

    it('renders instrument select', () => {
        render(
            <Formik initialValues={{}} onSubmit={noop}>
                <LocaleContext.Provider value={createLocaleContext(getStoreConfig())}>
                    <PaymentFormContext.Provider
                        value={{ paymentForm: getPaymentFormServiceMock() }}
                    >
                        <Field
                            name="instrumentId"
                            render={(field: FieldProps<string>) => (
                                <BraintreeAcceleratedCheckoutInstrumentSelect
                                    instruments={[getCardInstrument()]}
                                    onSelectInstrument={noop}
                                    onUseNewInstrument={noop}
                                    selectedInstrument={undefined}
                                    {...field}
                                />
                            )}
                        />
                    </PaymentFormContext.Provider>
                </LocaleContext.Provider>
            </Formik>,
        );

        expect(screen.queryByTestId('powered-by-paypal-connect-label')).not.toBeInTheDocument();
    });
});
