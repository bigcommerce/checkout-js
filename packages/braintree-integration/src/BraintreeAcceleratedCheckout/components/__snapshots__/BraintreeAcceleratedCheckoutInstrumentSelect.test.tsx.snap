// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BraintreeAcceleratedCheckoutInstrumentSelect renders instrument select 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="instrumentSelect"
        data-test="instrument-select"
      >
        <div
          class="dropdownTrigger"
        >
          <button
            class="instrumentSelect-button optimizedCheckout-form-select dropdown-button form-input"
            data-test="instrument-select-button"
            type="button"
          >
            <div
              class="instrumentSelect-details"
            >
              <div
                class="cardIcon-icon cardIcon-icon--default icon icon--medium"
              />
              <div
                class="instrumentSelect-card"
                data-test="instrument-select-button-last4"
              >
                Card ending in 4321
              </div>
              <div
                class="instrumentSelect-expiry"
                data-test="instrument-select-button-expiry"
              >
                Expires 02/2025
              </div>
              <div
                class="instrument-powered-by-icon"
              >
                <div
                  class="icon"
                >
                  <svg
                    height="24px"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      fill="#F1F1F1"
                      r="12"
                    />
                    <g>
                      <g>
                        <path
                          d="M8.6,12.9V9.5H7.3c-0.8,0-1.4,0.7-1.4,1.6v4.7c0,0.9,0.6,1.6,1.4,1.6h6.6c0.8,0,1.4-0.7,1.4-1.6v-1.3h-5.3
                    C9.3,14.5,8.6,13.8,8.6,12.9z"
                          fill="#313131"
                        />
                        <path
                          d="M16.7,6.5h-6.6c-0.8,0-1.4,0.7-1.4,1.6v1.3h5.3c0.8,0,1.4,0.7,1.4,1.6v3.4h1.3c0.8,0,1.4-0.7,1.4-1.6V8.2
                    C18.1,7.2,17.5,6.5,16.7,6.5z"
                          fill="#5B5B5B"
                        />
                      </g>
                      <path
                        d="M13.9,9.5H8.6v3.4c0,0.9,0.6,1.6,1.4,1.6h5.3v-3.4C15.4,10.2,14.7,9.5,13.9,9.5z"
                        fill="#DBDBDB"
                      />
                    </g>
                  </svg>
                </div>
              </div>
            </div>
          </button>
          <input
            name="instrumentId"
            type="hidden"
            value=""
          />
        </div>
        <div
          data-test="powered-by-paypal-connect-label"
        >
          PoweredByPaypalConnectLabel
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="instrumentSelect"
      data-test="instrument-select"
    >
      <div
        class="dropdownTrigger"
      >
        <button
          class="instrumentSelect-button optimizedCheckout-form-select dropdown-button form-input"
          data-test="instrument-select-button"
          type="button"
        >
          <div
            class="instrumentSelect-details"
          >
            <div
              class="cardIcon-icon cardIcon-icon--default icon icon--medium"
            />
            <div
              class="instrumentSelect-card"
              data-test="instrument-select-button-last4"
            >
              Card ending in 4321
            </div>
            <div
              class="instrumentSelect-expiry"
              data-test="instrument-select-button-expiry"
            >
              Expires 02/2025
            </div>
            <div
              class="instrument-powered-by-icon"
            >
              <div
                class="icon"
              >
                <svg
                  height="24px"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    fill="#F1F1F1"
                    r="12"
                  />
                  <g>
                    <g>
                      <path
                        d="M8.6,12.9V9.5H7.3c-0.8,0-1.4,0.7-1.4,1.6v4.7c0,0.9,0.6,1.6,1.4,1.6h6.6c0.8,0,1.4-0.7,1.4-1.6v-1.3h-5.3
                    C9.3,14.5,8.6,13.8,8.6,12.9z"
                        fill="#313131"
                      />
                      <path
                        d="M16.7,6.5h-6.6c-0.8,0-1.4,0.7-1.4,1.6v1.3h5.3c0.8,0,1.4,0.7,1.4,1.6v3.4h1.3c0.8,0,1.4-0.7,1.4-1.6V8.2
                    C18.1,7.2,17.5,6.5,16.7,6.5z"
                        fill="#5B5B5B"
                      />
                    </g>
                    <path
                      d="M13.9,9.5H8.6v3.4c0,0.9,0.6,1.6,1.4,1.6h5.3v-3.4C15.4,10.2,14.7,9.5,13.9,9.5z"
                      fill="#DBDBDB"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </div>
        </button>
        <input
          name="instrumentId"
          type="hidden"
          value=""
        />
      </div>
      <div
        data-test="powered-by-paypal-connect-label"
      >
        PoweredByPaypalConnectLabel
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
